sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -103,8 +119,26 @@ protected String visitIsNotNull(String v) {
     return v + " IS NOT NULL";
   }
 
+  protected String visitStringPredicate(String name, String l, String r) {
+    String value = r.replaceAll("'", "");
+    switch (name) {
+      case "STARTS_WITH":
+        return l + " LIKE '" + value + "%'";
+      case "ENDS_WITH":
+        return l + " LIKE '%" + value + "'";
+      default: // "CONTAINS"
```

Comment:
Let's match `CONTAINS` explicitly and the default case should throw exception, so that we won't make mistake when we add new string predicates later.

Reply:
OK